### ISSUE TITLE
Allow using sRGB as default color space on load

### ISSUE THREAD

**Issue by aaronscherzinger** [microsoft/DirectXTex#177]
I have recently encountered an issue when loading a png-file using the function `LoadFromWICMemory()` in DirectXTex.h / DirectXTexWIC.cpp. It seems that the png-file does not explicitly specify the color format. `LoadFromWICMemory()` therefore seems to assume a linear RGB format, while all tools I have tested seem to assume sRGB as a default.
It would be very useful to have a flag in order to switch the default format (i.e., behavior when there is no explicit color format given) to sRGB instead of linear RGB.
As a workaround, I added support for the `WIC_FLAGS_FORCE_SRGB` flag by adding the following code in the function `DecodeMetadata()`: `if (flags & WIC_FLAGS_FORCE_SRGB) sRGB = true;`
which might also be useful, but is of course a different behavior since it would ignore an explicitly set linear format.

**Comment by walbourn** [comment]
Every PNG written by WIC has ``sRGB`` in it by default, so I'm assuming you are talking about non-Windows tools, correct?

So far I have:

* ``WIC_FLAGS_FORCE_LINEAR`` which assumes the format is linear no matter what the metadata says
* ``WIC_FLAGS_FORCE_SRGB`` which assumes the format is sRGB no matter what the metadata says (although this only matters for DXGI formats that *have* SRGB equivalents)
* ``WIC_FLAGS_IGNORE_SRGB`` which ignores any metadata in the file, which because DXGI has only a few explicitly sRGB formats defined means it defaults to 'standard' without the ``_SRGB``.

Can you attach an example PNG that lacks any sRGB/gamma information?

**Comment by aaronscherzinger** [comment]
I have not created the PNG myself, so I am not sure which tool was used to write it, but it may very well have been a non-Windows tool. I have attached the image file, thanks for looking into it!

To give a little bit more context, loading image files is handled in different ways in our code base. One code path (purely in C++) loads the file "as is" in form of a binary buffer and hands it over to `LoadFromWICMemory()` (this is the one that "fails" in that it creates a linear RGB texture). We also have a code path used in a different application where the file is first loaded using C#'s Bitmap class, then saved to a buffer that is handed over to the C++ part, again using `LoadFromWICMemory()` - here, it seems that C# explicitly sets the sRGB information so that the texture is created correctly with the sRGB format.

Regarding the `WIC_FLAGS_FORCE_SRGB` flag, as far as I have seen this is ignored by `LoadFromWICMemory()` which is why as a test I added it in `DecodeMetaData()` (but the preferrable option would be to not force sRGB as we would like to still support linear image data).


![array_indexing](https://user-images.githubusercontent.com/50294132/82648763-6f610980-9c18-11ea-912b-0ddc6df3766a.png)
